models/blur/ethereum/blur_ethereum_events.sql

```diff
@@ -134,4 +134,74 @@ LEFT JOIN {{ ref('nft_ethereum_transfers') }} seller_fix ON seller_fix.block_tim
+, s.contract_address AS project_contract_address
+, get_json_object(s.offer[0], '$.token') AS nft_contract_address
+, NULL AS aggregator_name
+, NULL AS aggregator_address
+, s.evt_tx_hash AS tx_hash
+, tx.from AS tx_from
+, tx.to AS tx_to
+, CAST(0 AS DOUBLE) AS platform_fee_amount_raw
+, CAST(0 AS DOUBLE) AS platform_fee_amount
+, CAST(0 AS DOUBLE) AS platform_fee_amount_usd
+, CAST(0 AS DOUBLE) AS platform_fee_percentage
+, LEAST(get_json_object(s.consideration[0], '$.amount'), get_json_object(s.consideration[1], '$.amount')) AS royalty_fee_amount_raw
+, LEAST(get_json_object(s.consideration[0], '$.amount'), get_json_object(s.consideration[1], '$.amount'))/POWER(10, 18) AS royalty_fee_amount
+, pu.price*LEAST(get_json_object(s.consideration[0], '$.amount'), get_json_object(s.consideration[1], '$.amount'))/POWER(10, 18) AS royalty_fee_amount_usd
+, 100.0*LEAST(get_json_object(s.consideration[0], '$.amount'), get_json_object(s.consideration[1], '$.amount'))
+    /(get_json_object(s.consideration[0], '$.amount')+get_json_object(s.consideration[1], '$.amount')) AS royalty_fee_percentage
+, currency_tok.symbol AS royalty_fee_currency_symbol
+, CASE WHEN get_json_object(s.consideration[0], '$.recipient')!=s.recipient THEN get_json_object(s.consideration[0], '$.recipient')
+    ELSE get_json_object(s.consideration[1], '$.recipient')
+    END AS royalty_fee_receive_address
+, 'ethereum-blur-v1-' || s.evt_block_number || '-' || s.evt_tx_hash || '-' || s.evt_index AS unique_trade_id
+FROM {{ source('seaport_ethereum','Seaport_evt_OrderFulfilled') }} s
+INNER JOIN {{ source('ethereum', 'transactions') }} tx ON tx.block_number=s.evt_block_number
+    AND tx.hash=s.evt_tx_hash
+    {% if is_incremental() %}
+    AND tx.block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+LEFT JOIN tokens_ethereum.nft nft_tok ON nft_tok.contract_address=get_json_object(s.offer[0], '$.token')
+LEFT JOIN tokens_ethereum.erc20 currency_tok ON currency_tok.contract_address=get_json_object(s.consideration[0], '$.token')
+LEFT JOIN {{ ref('prices_usd_forward_fill') }} pu ON ((pu.contract_address=get_json_object(s.consideration[0], '$.token') AND pu.blockchain='ethereum')
+        OR (get_json_object(s.consideration[0], '$.token')='0x0000000000000000000000000000000000000000'  AND pu.blockchain IS NULL AND pu.contract_address IS NULL AND pu.symbol='ETH'))
+    AND pu.minute=date_trunc('minute', s.evt_block_time)
+    {% if is_incremental() %}
+    AND pu.minute >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+LEFT JOIN {{ ref('nft_ethereum_transfers') }} tr ON tr.block_time=s.evt_block_time
+    AND tr.contract_address=get_json_object(s.offer[0], '$.token')
+    AND tr.token_id=get_json_object(s.offer[0], '$.identifier')
+    AND tr.from=s.offerer
+    {% if is_incremental() %}
+    AND tr.block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+WHERE s.evt_block_time >= '2023-01-25'
```

Comment:
rather than hardcode this here, for best practice with jinja syntax, can we move this to not incremental block below (or via an else block)?